pkg/plugins/golang/deploy-image/v1alpha1/scaffolds/internal/templates/controllers/controller.go

```diff
@@ -220,6 +254,19 @@ func labelsFor{{ .Resource.Kind }}(name string) map[string]string {
 	return map[string]string{"type": "{{ lower .Resource.Kind }}", "{{ lower .Resource.Kind }}_cr": name}
 }
 
+func (r *{{ .Resource.Kind }}Reconciler) finalize{{ .Resource.Kind }}Finalizer(ctx context.Context, log logr.Logger, m *{{ .Resource.ImportAlias }}.{{ .Resource.Kind }}) error {
```

Comment:
teh deployment will be deleted we need to do an extra operation like raise en event